sensor/common/scannerclient/grpc_client.go

```diff
@@ -83,10 +183,74 @@ func (c *Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg
 
 	log.Debugf("Received image components from local Scanner for image %s", name)
 
-	return resp, nil
+	return &ImageAnalysis{
+		ScanStatus:   resp.GetStatus(),
+		ScanNotes:    resp.GetNotes(),
+		V1Components: resp.GetComponents(),
+	}, nil
+}
+
+// Close closes and cleanup the client connection.
+func (c *v2Client) Close() error {
+	return c.conn.Close()
+}
+
+func convertIndexReportToAnalysis(ir *v4.IndexReport) *ImageAnalysis {
+	var st scannerV1.ScanStatus
+	switch ir.GetState() {
+	case "Terminal", "IndexError":
+		st = scannerV1.ScanStatus_FAILED
+	case "IndexFinished":
+		st = scannerV1.ScanStatus_SUCCEEDED
+	default:
+		st = scannerV1.ScanStatus_ANALYZING
+	}
+	return &ImageAnalysis{
+		ScanStatus: st,
+		V4Contents: ir.GetContents(),
+	}
+}
+
+func (c *v4Client) GetImageAnalysis(ctx context.Context, image *storage.Image, cfg *types.Config) (*ImageAnalysis, error) {
+	imgSHA := utils.GetSHA(image)
+	if imgSHA == "" {
+		// TODO: ROX-19576: Is the assumption that image always have SHA correct?
+		return nil, fmt.Errorf("internal: V4 scanning of an image without SHA: %s",
+			image.GetName().GetFullName())
+	}
+	hashID := fmt.Sprintf("/v4/containerimage/%s", imgSHA)
+	opts := []grpc.CallOption{grpc.WaitForReady(true)}
+	ir, err := c.GetIndexReport(ctx, &v4.GetIndexReportRequest{HashId: hashID}, opts...)
```

Comment:
unrelated, but do you think `hashID` is redundant? I feel like `hash` or `id` alone is sufficient (perhaps ID is too generic and doesn't make it clear it is meant to be a hash). Also unrelated, do we want to enforce any kind of hash type (sha256, sha512, etc)?